src/CovidData/covidDataAPI.ts

```diff
@@ -0,0 +1,184 @@
+import Logger from "../logger"
+
+import { CovidData, CovidDatum } from "./covidData"
+
+const COVID_DATA_ENDPOINT =
+  "https://localcoviddata.com/covid19/v1/cases/covidTracking?state="
+const DAYS_IN_PAST_PARAM = "&daysInPast=7"
+
+const requestHeaders = {
+  "content-type": "application/json",
+  accept: "application/json",
+}
+
+export interface RequestSuccess {
+  kind: "success"
+  data: CovidData
+}
+
+type NetworkError =
+  | "InternalError"
+  | "JsonDeserialization"
+  | "NetworkConnection"
+  | "Unknown"
+
+export interface RequestFailure {
+  kind: "failure"
+  error: NetworkError
+}
+
+interface NetworkModel {
+  stateName: string
+  stateCode: string
+  historicData: NetworkData
+}
+
+type NetworkData = NetworkDatum[]
+
+type Date = string
+
+interface NetworkDatum {
+  date: Date
+  peoplePositiveCasesCt: number
+  peoplePositiveNewCasesCt: number
+  peopleNegativeCasesCt: number
+  peopleNegativeNewCt: number
+  peoplePendingCasesCt: number
+  peopleRecoveredCt: number
+  peopleDeathCt: number
+  peopleDeathNewCt: number
+  peopleHospCurrentlyCt: number
+  peopleHospNewCt: number
+  peopleHospCumlCt: number
+  peopleInIntnsvCareCurrCt: number
+  peopleInIntnsvCareCumlCt: number
+  peopleIntubatedCurrentlyCt: number
+  peopleIntubatedCumulativeCt: number
+}
+
+export const toCovidData = (networkModel: NetworkModel): CovidData => {
+  const toCovidDataDatum = (networkDatum: NetworkDatum): CovidDatum => {
+    const {
+      date,
+      peoplePositiveNewCasesCt,
+      peoplePositiveCasesCt,
+      peopleDeathCt,
+      peopleDeathNewCt,
+    } = networkDatum
+
+    return {
+      date,
+      positiveCasesNew: peoplePositiveNewCasesCt,
```

Comment:
Thoughts on swapping this so the assignment happens in the destructuring from `networkDatum`?